### DISCUSSION TITLE
Tuya motion sensors PIR high delay

### DISCUSSION THREAD

**Discussion by ghost** [Koenkk/zigbee2mqtt#22261] [Devices]
Hi,

I have Z2M 1.36.1 running on a RPI3 together with a Mosquitto Broker and NodeRed for the Logic. No Docker container. 
Coordinator is a Sonoff ZBDongle-P. 
~30 devices are connected: bulbs, plugs, temperature sensors, Xiaomi buttons and motion sensors of Xiaomi and Tuya.

From Tuya there are two different PIRs:
https://www.zigbee2mqtt.io/devices/809WZT.html#tuya-809wzt
https://www.zigbee2mqtt.io/devices/ZG-204ZL.html#tuya-zg-204zl
and from Xiaomi only these:
https://www.zigbee2mqtt.io/devices/RTCGQ01LM.html#xiaomi-rtcgq01lm

The Xiaomis work fine since years, now a few weeks ago I started also adding the Tuyas. Sadly,  the response time of the Tuyas is very high. If I trigger Xiaomi and Tuya in parallel, the Tuyas (no matter if with or without Lux) report ~1500ms after the Xiaomis. First I thought it could be the sensors, but even the once of a different type I have ordered have the same problem.

I can see the LED of the Tuyas blinking instantly, so it is not related to that they trigger too slow.
It does not matter how far away they are from the coordinator.
The latency is also visible in the Z2M log, not only in NodeRed.
All other devices work fine, f.e. a light bulb I set to on in Z2M almost does not have any latency. Same for the Xiaomi buttons, I can see the press with almost no latency.

As I did not find any reports of high latency problems with Tuya PIRs and Z2M on Google or here, it seems that this problem is not common.

Any ideas?

**Comment by Talixme** [comment]
Same problem
